launcher/ui/widgets/WideBar.cpp

```diff
@@ -133,7 +158,9 @@ void WideBar::insertSeparator(QAction* before)
         return;
 
     BarEntry entry;
-    entry.bar_action = QToolBar::insertSeparator(before);
+    entry.bar_action = new QAction("", this);
+    entry.bar_action->setSeparator(true);
+    insertAction(iter->bar_action, entry.bar_action);
```

Comment:
why is this change needed?

Reply:
it was adding the separator to the end of the toolbar instead of before the action

Reply:
that's because the old call was using `before` as the action to insert in relation to, i believe. It should use `iter->bar_action` instead, since that's the actual action in the WideBar.

Reply:
done